### ISSUE TITLE
Accessibility: Enable `accessibilityAudit` on different pages in our integration tests

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#33093]
## Problem statement

We recently added `accessibilityAudit` to our integration tests (https://github.com/sourcegraph/sourcegraph/pull/32895). It will test the current page for accessibility and fail with any errors.

We should aim to use this in all areas where we use `percySnapshot`
## Success criteria

We have enabled `accessibilityAudit` in all the areas where we use `percySnapshot` in the integration tests.

## Implementation details

 Here's how we should implement:

1. Add `accessibilityAudit` in the same place where we call `percySnapshot` in the integration tests.
2. Run the tests locally and observe any failures. See the above PR for example output.

To fix the failures, we should:

a. If it's simple or quick to fix, then make the changes on the branch. There should be relevant documentation attached to the test error that will help you work out how to fix this.
b. If it's complex, or will take a long time to fix. Add the `a11y-ignore` class to exclude the element from the audit, raise an issue specifically to solve this problem, leave a comment next to the class with a link to the issue.

Useful links:
[accessibilityAudit](https://github.com/sourcegraph/sourcegraph/pull/32895)
[Accessibility documentation to help debug](https://docs.sourcegraph.com/dev/background-information/web/accessibility)

## Time estimate

We should not spend more than **4 hours** per addition of `accessibilityAudit`. If we find ourselves taking more than that, we should create a separate issue for the complex problems.

**Comment by gitstart-sourcegraph** [comment]
@umpox We have some instances of `monaco-editor` breaking accessibility with `landmark-unique` rule, due to there being two `.monaco-status` `divs` with `role=complementary` but no `aria-label`. Here's an [existing issue](https://github.com/microsoft/monaco-editor/issues/2448) that highlights this, I believe this issue should be resolved in later versions of `monaco-editor` than we have at the moment. WDYT about spending some time to upgrade `monaco-editor` to version to resolve this, or an alternative solution perhaps?. Considering it's a dynamically generated element, disabling `accessibilityAudit` might be tricky.

**Comment by umpox** [comment]
@gitstart-sourcegraph It appears actually that bug [has not been fixed](https://sourcegraph.com/github.com/microsoft/vscode/-/blob/src/vs/base/browser/ui/aria/aria.ts?L34-37), so we'll have to add an additional workaround here.

Let's update this [line](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/shared/src/testing/accessibility.ts?L50%3A45=) to take an array of ignored classes, so we can still ignore even if it is dynamically added.

```
    const axe = new AxePuppeteer(page).exclude([
        ACCESSIBILITY_AUDIT_IGNORE_CLASS,
        '.monaco-status', // https://github.com/microsoft/monaco-editor/issues/2448
    ])
```

**Comment by raph941** [comment]
Thanks @umpox

**Comment by raph941** [comment]
Thank @umpox
